src/app.js

```diff
@@ -20,19 +24,34 @@ export default class App extends React.Component {
     window.addEventListener('beforeunload', () => {
       const myCards = JSON.stringify(this.state.cards)
       localStorage.setItem('cards', myCards)
+      const currentId = JSON.stringify(this.state.nextId)
+      localStorage.setItem('id', currentId)
     })
   }
-  addCard(card) {
-    const addCard = this.state.cards.slice()
-    addCard.push(card)
-    this.setState({cards: addCard})
+  submitHandler(card) {
+    if (this.state.view.path === 'edit') {
+      const changeCardIndex = this.state.cards.findIndex(curr => curr.cardId === card.cardId)
+      const before = this.state.cards.slice(0, changeCardIndex)
+      const after = this.state.cards.slice(changeCardIndex + 1)
+      const newCards = [...before, card, ...after]
+      this.setState({
+        cards: newCards
+      })
+      location.hash = '#list'
+    }
+    else if (this.state.view.path === 'new') {
+      const addCard = this.state.cards.slice()
+      addCard.push(card)
+      this.setState({cards: addCard,
+        nextId: this.state.nextId + 1})
+    }
   }
   render() {
     return (
       <React.Fragment>
         <Navbar/>
-        {this.state.view === 'new' && <Form submit={this.addCard}/>}
-        {this.state.view === 'list' && <CardList cards={this.state.cards}/>}
+        {(this.state.view.path === 'new' || this.state.view.path === 'edit') && <Form submit={this.submitHandler} nextId={this.state.nextId} cards={this.state.cards} params={this.state.view.params} mode={this.state.view.path}/>}
```

Comment:
This conditional rendering stuff is really hard to read. Try extracting a method that knows which view to return.